### PR TITLE
feat: distributed session keys

### PR DESCRIPTION
**Pull request by joepegler**
## PR-Codex overview
This PR introduces a new feature related to wallet provider service client configuration and adds constants for session management.

### Detailed summary
- Added `ClientConfig` for wallet provider service client setup
- Updated constants related to session management
- Modified error handling in `getAAError`
- Updated types in `walletProviderServiceClientInterface`
- Added functions for creating viem accounts in `viemSigner`

> The following files were skipped due to too many changes: `src/modules/walletprovider-sdk/walletProviderServiceClient.ts`, `src/modules/index.ts`, `tests/playground/write.test.ts`, `src/modules/utils/Types.ts`, `examples/CREATE_AND_USE_A_SESSION.md`, `package.json`, `src/modules/walletprovider-sdk/authentication.ts`, `src/account/utils/Types.ts`, `src/modules/walletprovider-sdk/EOAauthentication.ts`, `src/modules/utils/Helper.ts`, `src/modules/walletprovider-sdk/networkSigner.ts`, `src/modules/sessions/abi.ts`, `src/modules/sessions/sessionSmartAccountClient.ts`, `tests/modules/write.test.ts`, `src/modules/DANSessionKeyManagerModule.ts`, `src/account/BiconomySmartAccountV2.ts`, `src/modules/sessions/dan.ts`, `src/modules/walletprovider-sdk/index.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/account/BiconomySmartAccountV2.ts [hunk 1]
```diff
@@ -1487,56 +1508,95 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
...
-   *      verificationGasLimitOffsetPct: 25, // 25% increase for the already estimated gas limit
-   *      preVerificationGasOffsetPct: 10 // 10% increase for the already estimated gas limit
-   *     }
-   * });
-   * const { transactionHash, userOperationReceipt } = await wait();
-   *
+   * ```
+   * 
+   * 
+   *
```
Comment (AmanRaj1608) [inline src/account/BiconomySmartAccountV2.ts:1514]: Should remove this?

File: src/modules/walletprovider-sdk/build/walletProviderServiceClientInterface.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import type { UserAuthentication } from './authentication';
+import type { KeygenResponse, KeygenSetupOpts, SignResponse, SignSetupOpts } from './networkSigner';
+/**
+ * The config used to create Wallet Provider Service backend client.
+ * Please refer to {@link https://shipyard.rs/silencelaboratories/crates/wallet-provider-service | example backend service}
```
Comment (AmanRaj1608) [inline src/modules/walletprovider-sdk/build/walletProviderServiceClientInterface.ts:5]: I don't think this is public accessible url

File: tests/modules/write.test.ts [hunk 3]
```diff
@@ -1451,4 +1460,209 @@ describe("Modules:Write", () => {
...
+    // Assume we know that the relevant session leaf to the transaction is the last one...
+    const allLeaves = await memoryStore.getAllSessionData();
+    const relevantLeaf = allLeaves[allLeaves.length - 1];
+    const sessionID = relevantLeaf.sessionID;
+
+    const nftBalanceBefore = await checkBalance(smartAccountAddress, nftAddress);
+    // Now use the sessionID and the DAN module info to send the transaction
+    const { wait: waitForMint } = await unconnectedSmartAccount.sendTransaction(nftMintTx, {
+      ...withSponsorship,
+      params: { sessionID }
```
Comment (livingrockrises) [inline tests/modules/write.test.ts:1656]: wouldnt we need to send here danModuleInfo stuff?

Comment (joepegler) [reply tests/modules/write.test.ts:1656]: Not any more - I pluck it out of the sessionStorageClient later using the leaf (found with the sessionID)

File: tests/modules/write.test.ts [hunk 4]
```diff
@@ -1451,4 +1460,209 @@ describe("Modules:Write", () => {
...
+    const sessionsModule = await DANSessionKeyManagerModule.create({
+      smartAccountAddress,
+      sessionStorageClient: memoryStore
+    })
+
+    // Set the ttl for the session 
+    const duration = 60 * 60
+
+    // Get the session key from the dan network
+    const danSessionData = await getDANSessionKey(
```
[the remark this answers is not in the document]
Comment (joepegler) [reply tests/modules/write.test.ts]: Added the return type

**Review by AmanRaj1608 — APPROVED** [review]

**Review by livingrockrises — APPROVED** [review]
LGTM

**Comment by joepegler** [comment]
Closing until silence labs come out of testnet phase.

**Pull request closed without merging**